### PR TITLE
[helidon-maven-plugin] Improve module support

### PR DESCRIPTION
**Pull request by tvallin**
See https://github.com/helidon-io/helidon-build-tools/issues/771

Signed-off-by: tvallin <thibault.vallin@oracle.com>

### CODE REVIEW

**Comment by romain-grecourt** [comment]
On the new classpath / module path bean, it would be nice to allow passing additional entries.
e.g.

```
<modulePath>
    <additionalEntries>
        <entry>foo.jar</entry>
        <!-- ... -->
    </additionalEntries>
</modulePath>
```

**Comment by tvallin** [comment]
So it is additional entries to the derived class/module-path right ?

File: helidon-maven-plugin/src/main/java/io/helidon/build/maven/graal/FilteringComponent.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import io.helidon.build.util.SourcePath;
+
+/**
+ * A configuration filtering component using excludes and includes list.
+ */
+public class FilteringComponent {
```
Comment (romain-grecourt) [inline helidon-maven-plugin/src/main/java/io/helidon/build/maven/graal/FilteringComponent.java:28]: How about we name this class something else, maybe just PathComponent ?

Comment (tvallin) [reply helidon-maven-plugin/src/main/java/io/helidon/build/maven/graal/FilteringComponent.java]: Done

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**